pkg/registry/generic/registry/store.go

```diff
@@ -249,6 +250,27 @@ func (e *Store) Create(ctx api.Context, obj runtime.Object) (runtime.Object, err
 		if accessor.GetDeletionTimestamp() != nil {
 			msg := &err.(*kubeerr.StatusError).ErrStatus.Message
 			*msg = fmt.Sprintf("object is being deleted: %s", *msg)
+			userAgent, _ := api.UserAgentFrom(ctx)
+			if !strings.Contains(userAgent, "kubectl") {
+				return nil, err
+			}
+			// example userAgent string: kubectl-1.3/v1.3.8 (linux/amd64) kubernetes/e328d5b
+			userAgent = strings.Split(userAgent, " ")[0]
+			subs := strings.Split(userAgent, "/")
+			if len(subs) != 2 {
+				return nil, err
+			}
+			kubectlVersion, versionErr := version.Parse(subs[1])
+			if versionErr != nil {
+				return nil, err
+			}
+			if kubectlVersion.GTE(version.MustParse("v1.4.0")) {
+				return nil, err
+			}
+			if e.QualifiedResource.Resource != "replicationcontrollers" {
+				return nil, err
+			}
+			*msg = fmt.Sprintf("Note: if you are using \"kubectl rolling-update\" and your kubectl version is older than v1.4.0, your rolling-update is probably failed. Your pods are successfully updated, but the name of the RC is its original name plus a hash value. If you want to rename the RC, please use a kubectl newer than v1.4.0, then run rolling-update to change the name. : %s", *msg)
```

Comment:
I think the 'mini-postmortem' would contain mostly the statement that the problem appeared because no one from people working on this feature knew anything about kubectl, and it wasn't caught by our upgrade tests because we didn't have any that were passing ever...